src/workerd/io/io-own.h

```diff
@@ -195,6 +231,9 @@ class DeleteQueuePtr: public kj::Own<DeleteQueue> {
   ~DeleteQueuePtr() noexcept(false) {
     auto ptr = get();
     if (ptr != nullptr) {
+      // The delete queue state may include a kj::CrossThreadPromiseFulfiller that
```

Comment:
Please only delete the CrossThreadPromiseFulfiller itself in AllowAsyncDestructorScope. We might also be deleting scheduled action functions here, and we should keep the restriction that those should not contain KJ objects.